crates/bevy_animation/src/interpolate/utils.rs

```diff
@@ -0,0 +1,125 @@
+use std::ops::{Add, Div, Mul, Sub};
+
+use bevy_math::{Vec2, Vec3};
+use ultraviolet::{f32x4, f32x8, Vec4x4, Vec4x8};
+use wide::{i32x4, i32x8};
+
+// TODO: Add Quat support to ultraviolet?
+
+/// Wide quaternion
+#[derive(Debug, Clone)]
+pub struct Quatx4(pub Vec4x4);
+
+/// Wide quaternion
+#[derive(Debug, Clone)]
+pub struct Quatx8(pub Vec4x8);
+
+// macro_rules! wide_quat {
+//     ($t:tt, $) => {
+//         impl $t {
+//             pub fn identity() -> $t {
+//                 $t::splat(Vec4::new(0.0, 0.0, 0.0, 0.1))
+//             }
+//         }
+//     };
+// }
+
+// wide_quat!(Quat4);
+// wide_quat!(Quat8);
+
+/// Used to change the blending semantics of a `Vec2`
+#[derive(Debug, Clone)]
+pub struct Scale2(pub Vec2);
+
+/// Used to change the blending semantics of a `Vec3`
+#[derive(Debug, Clone)]
+pub struct Scale3(pub Vec3);
+
+/// Quake 3 fast inverse sqrt
+///
+/// Took this one from piston: https://github.com/PistonDevelopers/skeletal_animation
+#[inline(always)]
+pub fn inv_sqrt(x: f32) -> f32 {
+    let x2: f32 = x * 0.5;
+    let mut y: f32 = x;
+
+    let mut i: i32 = unsafe { std::mem::transmute(y) };
```

Comment:
f32::to_bits is safe

Reply:
Good catch `y.to_bits() as i32` should yield the same result

Reply:
Good catch `y.to_bits() as i32` should yield the same result